src/perm/builder/pow.rs

```diff
@@ -50,12 +49,25 @@ where
             return Permutation::id();
         }
 
-        MultiJoin::new(
-            std::iter::repeat(self.perm.clone())
-                .take(self.power.abs() as usize)
-                .map(|p| p.collapse()),
-        )
-        .collapse()
+        pow(self.perm.collapse(), self.power as usize)
+    }
+}
+
+fn pow(perm: Permutation, x: usize) -> Permutation {
```

Comment:
This function could perhaps use better documentation. It isn't doing anything particularly fancy, but could be good to explain the optimisation.

Reply:
Resolved this now!